File: docs/code-examples/all/mesh3d_indexed.py
"""Log a simple colored triangle."""
import rerun as rr

rr.init("rerun_example_mesh3d_indexed", spawn=True)

rr.log(
    "triangle",
    rr.Mesh3D(
        vertex_positions=[[0.0, 1.0, 0.0], [1.0, 0.0, 0.0], [0.0, 0.0, 0.0]],
        vertex_normals=[0.0, 0.0, 1.0],
        vertex_colors=[[0, 0, 255], [0, 255, 0], [255, 0, 0]],
        indices=[2, 1, 0],
    ),
)

File: docs/snippets/all/archetypes/mesh3d_indexed.cpp
// Log a simple colored triangle.

#include <rerun.hpp>

#include <vector>

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_mesh3d_indexed");
    rec.spawn().exit_on_failure();

    const rerun::Position3D vertex_positions[3] = {
        {0.0f, 1.0f, 0.0f},
        {1.0f, 0.0f, 0.0f},
        {0.0f, 0.0f, 0.0f},
    };
    const rerun::Color vertex_colors[3] = {
        {0, 0, 255},
        {0, 255, 0},
        {255, 0, 0},
    };

    rec.log(
        "triangle",
        rerun::Mesh3D(vertex_positions)
            .with_vertex_normals({{0.0, 0.0, 1.0}})
            .with_vertex_colors(vertex_colors)
            .with_triangle_indices({{2, 1, 0}})
    );
}
